pkg/manager/server.go

```diff
@@ -0,0 +1,44 @@
+package manager
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"golang.org/x/sync/errgroup"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+// Server holds details for a manager server.
+type Server struct {
+	manager manager.Manager
+	cancel  context.CancelFunc
+	logger  logr.Logger
+}
+
+// NewManagerServer returns a new manager server.
+func NewManagerServer(logger logr.Logger, manager manager.Manager) *Server {
+	return &Server{
+		logger:  logger,
+		manager: manager,
+	}
+}
+
+// Start starts the manager server.
+func (s *Server) Start(ctx context.Context) error {
+	shutdownContext, cancel := context.WithCancel(context.Background())
+	s.cancel = cancel
+	g, _ := errgroup.WithContext(ctx)
```

Comment:
Should this use the returned context?

Reply:
Nope, because the shutdown context needs to be separate from that. Yet, the passed in context is still important, because if anything else using that context from the manager dies, the context needs to be able to signal everything else that it died. So if one of the servers that the handler handles dies, everyone is notified and restarted. The assumption is that these are co-dependent servers. Now, if that is not the case, then just don't use the passed in context. Right now, the returned context here, has no use, as the server has its own shutdown context which it needs to keep track of separately to call with `Stop` to be conformant with the Service interface.